File: slides/omp/code/sum_serial.cc
#include <cstdio>
using std::printf;
int main()
{
 const long n = 100000;
 double v[n];
 for (int i = 0; i < n; ++i)
  v[i] = (double)i;
 // do the sum
 double s = 0.0;
#pragma omp parallel
{
#pragma omp for
 for (int i = 0; i < n; ++i)
#pragma omp critical
   s += v[i];
} // end omp parallel
 double s_ref = 0.5*(n*n - n);
 printf("sum is %f, "
  "expected %f\n", s, s_ref);
}

File: slides/omp/code/sum_serial.f90
program sum_serial
implicit none
integer*8, parameter :: &
 n = 100000
integer :: i
real*8 :: s, s_ref, v(n)
do i = 1, n
  s = s + (i-1)
end do
s_ref = 0.5_8*(n*n- n)
print *, "sum is ", s, &
         " expected ", s_ref
end program sum_serial
